pkg/types/chain_writer.go

```diff
@@ -10,9 +10,9 @@ import (
 type ChainWriter interface {
 	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// - `args` specifies input parameters to the contract call.
+	// - `args` should be any object which maps a set of method param into the contract and method specific method params.
 	// - `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
```

Comment:
can you add to the description of the PR why UUID doesn't work?

Reply:
UUID does work, and it's fine to send a UUID. It's just that it doesn't have to be enforced.